bigchaindb/commands/utils.py

```diff
@@ -74,12 +74,50 @@ def start_logging(args):
     return start_logging
 
 
+def _convert(value, default=None, convert=None):
+    def convert_bool(value):
+        if value.lower() in ('true', 't', 'yes', 'y'):
+            return True
+        if value.lower() in ('false', 'f', 'no', 'n'):
+            return False
+        raise ValueError('{} cannot be converted to bool'.format(value))
+
+    if value == '':
+        value = None
+
+    if convert is None:
+        if default is not None:
+            convert = type(default)
+        else:
+            convert = str
+
+    if convert == bool:
+        convert = convert_bool
+
+    if value is None:
+        return default
+    else:
+        return convert(value)
+
+
```

Comment:
Perhaps, just a question of personal preferences, but I would simplify the above in a few ways. As an example, a possible simplification could yield:

```python
def _convert_bool(value):
    if value.lower() in ('true', 't', 'yes', 'y'):
        return True
    if value.lower() in ('false', 'f', 'no', 'n'):
        return False
    raise ValueError('{} cannot be converted to bool'.format(value))


def _convert(value, default=None, convert=None):
    if value == '':
        return default

    if convert is None and default is None:
        return value

    if convert is None and default is not None:
        convert = type(default)

    if convert == bool:
        convert = _convert_bool

    return convert(value)
```